desc/compute/data_index.py

```diff
@@ -75,7 +80,63 @@ def _decorator(func):
             "coordinates": coordinates,
             "dependencies": deps,
         }
-        data_index[name] = d
+        for p in parameterization:
+            flag = False
+            for base_name, aliases in _aliases.items():
+                if p in aliases or p == base_name:
+                    data_index[base_name][name] = d.copy()
+                    flag = True
+            if not flag:
+                raise ValueError(
+                    f"Can't register function with unknown parameterization: {p}"
+                )
         return func
 
     return _decorator
+
+
+# we register things by their full module.class name, but sometimes we might refer
+# to it via import path or just class name
+# also allows us to handle subclasses whos data_index stuff should inherit
+# from parent classes.
+# This is the least bad solution I've found, since everything else requires
+# crazy circular imports
+# could maybe make this fancier with a registry of compute-able objects?
+_aliases = {
+    "desc.equilibrium.equilibrium.Equilibrium": [
+        "desc.equilibrium.Equilibrium",
+        "Equilibrium",
+    ],
+    "desc.geometry.curve.FourierRZCurve": [
```

Comment:
Unsure if I understand the aliases, so if I have a compute function that is specific to just FourierRZCurve, but not to XYZcurve (from the other PR I have #494 ), for example it is compute_coordinates, which for FourierRZCurve uses transforms, while XYZCurve uses splines. 
Is the idea that if I put as a paremetrization for my compute function decorator `XYZCurve`, then it will be able to be used by `XYZCurve` objects but not `FourierRZCurve` objects?
But if I say the parametrization for a compute function is `Curve`, then it will be allowed to be used with any subclass of `Curve` (according to our aliases here)?

Reply:
Yeah exactly. This is basically just to keep track of class inheritance without actually having to import all the classes to do regular `issubclass` and `isinstance` checks. In #583 you can see that there are multiple implementations for curve `x` (for each parameterization) but stuff like the frenet frame, curvature etc there is one shared function that works for any `Curve`

Reply:
I'll clean it up a bit to be just dealing with inheritance rather than also parsing different paths to the same class, we should just stick to the canonical path.